lib/final_destination.rb

```diff
@@ -36,7 +36,15 @@ def initialize(url, opts = nil)
     @force_get_hosts = @opts[:force_get_hosts] || []
     @opts[:max_redirects] ||= 5
     @opts[:lookup_ip] ||= lambda { |host| FinalDestination.lookup_ip(host) }
-    @ignored = [Discourse.base_url_no_prefix] + (@opts[:ignore_redirects] || [])
+
+    @ignored = @opts[:ignore_hostnames] || []
+    ([Discourse.base_url_no_prefix] + (@opts[:ignore_redirects] || [])).each do |url|
+      url = uri(url)
+      if url.present? && url.hostname
```

Comment:
Let's add a test case for this so that it becomes clearer why this condition is required.